src/test/java/nc/students/ayaz/controllers/UserControllerTest.java

```diff
@@ -0,0 +1,57 @@
+package nc.students.ayaz.controllers;
+
+import nc.students.ayaz.model.User;
+import nc.students.ayaz.model.Video;
+import nc.students.ayaz.repositories.UserRepository;
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+public class UserControllerTest {
+
+    @Autowired
+    private UserRepository repository;
+
+    @Autowired
+    private TestRestTemplate restTemplate;
+
+    @Test
+    public void shouldAddUserToRepositoryWhenPOSTrequestSent() {
+        ResponseEntity<String> response = restTemplate.postForEntity("/users/Ayaz", "", String.class);
+
+        assertTrue(repository.contains("Ayaz"));
+        assertEquals(HttpStatus.CREATED, response.getStatusCode());
```

Comment:
&#x1F34E; **Issue**
I would make sure that status code is correct first. Cause it will be easier to understand what the problem is for certain class of problems (test halts when assertion error is thrown).